quantlib/test/test_bondfunctions.py

```diff
@@ -0,0 +1,196 @@
+            maturity_date,
+            Period(Semiannual),
+            UnitedStates(market=GOVERNMENTBOND),
+            Unadjusted,
+            Unadjusted,
+            Backward,
+            False);
+
+
+        bond = FixedRateBond(
+            settlement_days,
+		    face_amount,
+		    fixed_bond_schedule,
+		    [coupon_rate],
+            ActualActual(Bond),
+		    Unadjusted,
+            redemption,
+            issue_date
+        )
+
+        bfs=bf.BondFunctions()
+        d=bfs.startDate(bond)
+        bfs.display()
+        zspd=bfs.zSpread(bond,100.0,flat_term_structure,Actual365Fixed(),
+        Compounded,Semiannual,settlement_date,1e-6,100,0.5)
+        
+        
+              
+        #Also need a test case for a PiecewiseTermStructure...                
+        depositData = [[ 1, Months, 4.581 ],
+                       [ 2, Months, 4.573 ],
+                       [ 3, Months, 4.557 ],
+                       [ 6, Months, 4.496 ],
+                       [ 9, Months, 4.490 ]]
+
+        swapData = [[ 1, Years, 4.54 ],
+                    [ 5, Years, 4.99 ],
+                    [ 10, Years, 5.47 ],
+                    [ 20, Years, 5.89 ],
+                    [ 30, Years, 5.96 ]]
+
+        rate_helpers = []
+
+        end_of_month = True
+        for m, period, rate in depositData:
+            tenor = Period(m, Months)
+
+            helper = DepositRateHelper(rate/100, tenor, settlement_days,
+                     calendar, ModifiedFollowing, end_of_month,
+                     Actual360())
+
+            rate_helpers.append(helper)
+
+        liborIndex = Libor('USD Libor', Period(6, Months), settlement_days,
+                           USDCurrency(), calendar, Actual360(),
+                           YieldTermStructure(relinkable=False))
+
+        spread = SimpleQuote(0)
+        fwdStart = Period(0, Days)
+
+        for m, period, rate in swapData:
+
+            helper = SwapRateHelper.from_tenor(
+                rate/100, Period(m, Years), calendar, Annual, Unadjusted, Thirty360(), liborIndex,
+                spread, fwdStart
+            )
+
+            rate_helpers.append(helper)
+
+        ts_day_counter = ActualActual(ISDA)
+        tolerance = 1.0e-15
+
+        ts = PiecewiseYieldCurve(
+            'discount', 'loglinear', settlement_date, rate_helpers,
+            ts_day_counter, tolerance)   
+
+        pyc_zspd=bfs.zSpread(bond,102.0,ts,ActualActual(ISDA),
+        Compounded,Semiannual,settlement_date,1e-6,100,0.5)                     
+
+        pyc_zspd_disco=bfs.zSpread(bond,95.0,ts,ActualActual(ISDA),
+        Compounded,Semiannual,settlement_date,1e-6,100,0.5)
+                                                          
+        # tests
+        #self.assertTrue(Date(27, January, 2011), bond.issue_date)
+        #self.assertTrue(Date(31, August, 2020), bond.maturity_date)
+        #self.assertTrue(settings.evaluation_date, bond.valuation_date)
+
+        print settings.evaluation_date
```

Comment:
! that code does not compile on Python 3 (see travis-ci failure). You should have:
`from __future__ import print_function` 

and then use only the print function:

`      print(d)
       print ('Principal: {}'.format(face_amount))
...
`